types/algoliasearch/index.d.ts

```diff
@@ -981,7 +981,7 @@ declare namespace algoliasearch {
   }
   interface BrowseResponse {
     cursor?: string;
-    hits: {}[];
+    hits: {objectID: string}[];
```

Comment:
This type is too restrictive. You can take a look at the [`Hit` type](https://github.com/algolia/instantsearch.js/blob/65911504b9ae0e2596c75f0ae74133985dc79850/src/types/instantsearch.ts#L46-L71) we created inside InstantSearch.js (which relies on `algoliasearch`). You can ignore the two `__position` and `__queryID` keys which are internal to InstantSearch.

[See Algolia response reference →](https://www.algolia.com/doc/api-reference/api-methods/search/#method-response-hits)

Reply:
The proposed change is less restrictive than what is currently implemented.
If additional work is desired, is it possible to do it in a follow-up PR?

*Note on the linked code above:*
I am very against adding the following line:
```ts
[attribute: string]: any;
```
It breaks down most of the typesafety in a way that is not reversible by the developer using it.

If anything it should be:
```ts
[attribute: keyof any]?: unknown;
```
Which requires correct narrowing into the type system before a value is accessible.

Reply:
that's a very good idea, I'd accept that @squirly,  thanks again!